lib/withLoggedInUser.js

```diff
@@ -93,10 +103,32 @@ const withLoggedInUser = WrappedComponent => {
           throw new Error('Invalid token');
         }
 
-        const newToken = await maybeRefreshAccessToken(token, twoFactorAuthenticatorCode);
-        if (newToken.code === 401) {
-          throw new Error(newToken.message);
-        } else if (!newToken) {
+        /**
+         * If we're logging in with a token, 2 things may happen:
+         * 1 It has scope of 'login' and there is no two factor authentication code passed into
+         * the function. We should call maybeRefreshAccessToken. If the user has no 2FA enabled,
+         * we exchange for a long term token and log in. If they do have 2FA enabled, we prompt
+         * them to enter the 6-digit code.
+         * 2 This function is then called again and token has scope 'twofactorauth' and we also
+         * get the code, so we call exchangeTwoFactorAuthenticationToken.
+         */
+        let newToken;
+        if (twoFactorAuthenticatorCode) {
+          newToken = await exchangeTwoFactorAuthenticationToken(token, twoFactorAuthenticatorCode);
+          if (newToken.code === 401) {
+            // if we receive a 401 error from the API this means the 2FA token has failed
+            throw new Error(newToken.message);
+          }
+        } else {
+          // Used for the first exchange of the login token
+          newToken = await maybeRefreshAccessToken(token);
+          const decodedNewToken = jwt.decode(newToken);
+          if (decodedNewToken.scope === 'twofactorauth') {
+            throw Error('Two-factor authentication is enabled on this account. Please enter the code');
```

Comment:
Does that works? It's supposed to be `new Error`

Reply:
I will change that

Reply:
It might work but this is not correct code, so yes please change.

https://stackoverflow.com/questions/9156176/what-is-the-difference-between-throw-new-error-and-throw-someobject

Reply:
Ok I'll do that an also update the existing code that also has `throw Error` instead of `throw new Error`